codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -147,13 +136,12 @@ public void operationComplete(Future<Void> future) throws Exception {
         // paths iterating over the active streams.
         if (activeStreams.allowModifications()) {
             while (itr.hasNext()) {
-                Http2Stream stream = itr.next().value();
```

Comment:
So I could add a package private method `close(Iterator<?> itr)` on `DefaultStream`. This `itr` is used to do the removal from `streamMap` if present to prevent any chance of concurrent modification.